js/src/UXCapture.js

```diff
@@ -0,0 +1,78 @@
+import ExpectedMark from "./ExpectedMark";
+import View from "./View";
+
+export default class UXCapture {
+  static attachTo(window) {
+    // allow running in Node.js environment
+    if (typeof window === "undefined") {
+      throw new Error("Must provide a valid window object");
+    }
+
+    // already have UX Capture object defined, reuse it
+    if (typeof window.UX !== "undefined") {
+      return window.UX;
+    }
+
+    // assign singleton instance to window object
+    window.UX = new UXCapture(window);
+
+    return window.UX;
+  }
+
+  constructor() {
+    this.onMark = null;
+    this.onMeasure = null;
```

Comment:
Not really, because it gets constructed automatically by the library when it attaches it to the window and then user of the library configures it later.